### PR TITLE
Added mempool unit test

### PR DESCRIPTION
**Pull request by natefitzgerald**
Hey this is my first contribution to this project. I'm planning on filling out at least as many of the mempool tests as I can. I thought I'd do a single one first to make sure I didn't mess up anything. Let me know if there's anything I can do better otherwise look for a PR with more of these in the next couple days.

* Added body for AcceptToMemoryPool_TxAncestorsConflictSpend_ReturnsFalseAsync

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs [hunk 1]
```diff
@@ -564,10 +564,49 @@ public void AcceptToMemoryPool_TxTooManyAncestors_ReturnsFalse()
...
+            Transaction conflictingTx = txBuilder
+               .AddCoins(coin)
+               .AddKeys(miner)
+               .Send(bob, "10.00")
+               .SendFees("0.001")
+               .SetChange(miner)
+               .BuildTransaction(true);
+
+            //Mempool should reject the second transaction
+            Assert.False(await validator.AcceptToMemoryPool(state, conflictingTx), $"Transaction: {nameof(conflictingTx)} should have failed mempool validation.");
```
Comment (noescape00) [inline src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs]: add `.ConfigureAwait(false)`

File: src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs [hunk 2]
```diff
@@ -564,10 +564,49 @@ public void AcceptToMemoryPool_TxTooManyAncestors_ReturnsFalse()
...
+               .AddCoins(coin)
+               .AddKeys(miner)
+               .Send(bob, "10.00")
+               .SendFees("0.001")
+               .SetChange(miner)
+               .BuildTransaction(true);
+            MempoolValidationState state = new MempoolValidationState(false);
+
+            //Mempool should accept it, there's nothing wrong
+            Assert.True(await validator.AcceptToMemoryPool(state, originalTx), $"Transaction: {nameof(originalTx)} failed mempool validation.");
```
Comment (noescape00) [inline src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs]: add `.ConfigureAwait(false)`

For everything that can be awaited we specify that continuation can be done on other synchronization context. 
It's not that important in tests but in general it should be used everywhere in the codebase

File: src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs [hunk 3]
```diff
@@ -564,10 +564,49 @@ public void AcceptToMemoryPool_TxTooManyAncestors_ReturnsFalse()
...
+            string dataDir = Path.Combine("TestData", nameof(MempoolValidatorTest), nameof(this.AcceptToMemoryPool_TxAncestorsConflictSpend_ReturnsFalseAsync));
+            Directory.CreateDirectory(dataDir);
+
+            BitcoinSecret miner = new BitcoinSecret(new Key(), Network.RegTest);
+            ITestChainContext context = await TestChainFactory.CreateAsync(Network.RegTest, miner.PubKey.Hash.ScriptPubKey, dataDir);
+            IMempoolValidator validator = context.MempoolValidator;
+            BitcoinSecret bob = new BitcoinSecret(new Key(), Network.RegTest);
+            TransactionBuilder txBuilder = new TransactionBuilder();
+
+            //Create Coin from first txn on chain
```
Comment (noescape00) [inline src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs]: typo `txn` => `tx`

**Comment by noescape00** [comment]
Good job, @natefitzgerald!

Btw a small tip: it's better to create a separate branch for each PR instead of using `master` so you can simultaneously work on several PRs instead of waiting for previous ones to get merged.

I personally have a small bash script: 

```
REPO_DIR='C:/Users/user/Desktop/StratisRepository'
cd $REPO_DIR
git checkout master
git pull https://github.com/stratisproject/StratisBitcoinFullNode.git master
git push origin master
$SHELL
```

I start it every time before creating a new branch for the PR to keep new stuff as close to the master's tip as possible to avoid merging issues. 
So my fullnode's master is only used for pulling stuff from the main repo. 
Otherwise you'd have to hard reset & pull & push --force every time you'd like to sync with the main repo.

File: src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs [hunk 4]
```diff
@@ -564,10 +564,49 @@ public void AcceptToMemoryPool_TxTooManyAncestors_ReturnsFalse()
         }
 
         [Fact]
-        public void AcceptToMemoryPool_TxAncestorsConflictSpend_ReturnsFalse()
+        public async void AcceptToMemoryPool_TxAncestorsConflictSpend_ReturnsFalseAsync()
         {
             // TODO: Execute failure cases for CheckAncestors
             // - conflicting spend transaction
+
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs]: You can now remove the `TODO:`  or are there more scenarios in this test coming?

**Comment by dangershony** [comment]
Thanks you @natefitzgerald looking forward for more commits. please fix the minor comments and lets push this.

**Comment by dangershony** [comment]
Reminder to @natefitzgerald to finish this PR, if you don't have the time one of us can make the last fix and push.

**Comment by bokobza** [comment]
@noescape00 @dangershony I've changed this based on your comments. Please check it out :-)

File: src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs [hunk 5]
```diff
@@ -544,10 +544,48 @@ public void AcceptToMemoryPool_TxTooManyAncestors_ReturnsFalse()
...
+            string dataDir = GetTestDirectoryPath(this);
+
+            BitcoinSecret miner = new BitcoinSecret(new Key(), Network.RegTest);
+            ITestChainContext context = await TestChainFactory.CreateAsync(Network.RegTest, miner.PubKey.Hash.ScriptPubKey, dataDir);
```
Comment (noescape00) [inline src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs]: .ConfigureAwait(false)

Comment (noescape00) [reply src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs]: Other than that- looks good to me.

**Pull request merged**